src/EventStore.Core/Services/Storage/ReaderIndex/IndexReader.cs

```diff
@@ -437,7 +437,7 @@ public class IndexReader<TStreamId> : IndexReader, IIndexReader<TStreamId> {
 		}
 
 		private bool OriginalStreamExists(TFReaderLease reader, TStreamId metaStreamId) {
-			if (_systemStreams.IsSystemStream(metaStreamId)) {
```

Comment:
i wonder if this was just a bug or if theres some reason it was this way

Reply:
looks like a bug to me too.
Based on the PR that introduced it: https://github.com/EventStore/EventStore/pull/709, it looks like the error has been repeated here as well (.IsSystemStream instead of IsMetaStream):
https://github.com/EventStore/EventStore/blob/master/src/EventStore.Core/Services/Storage/StorageReaderWorker.cs#L280

Reply:
i've fixed it in this commit: 5eadcf62f3d061da9716782cda2879b0820d5691